forum/tests.py

```diff
@@ -13,63 +26,305 @@
+        }
+        request = self.factory.post(
+            '/advising/forum/Fall/2019/', data=data, format='json')
+        response = get_response_for_semester(self, request, self.advisor.user)
+        self.assertEquals(response.status_code, status.HTTP_201_CREATED)
+        comment = Comment.objects.get(
+            transcript=self.transcript, author=self.advisor)
+        self.assertEquals(content, comment.content)
+
+    def test_add_advisor(self):
+        setUpTranscriptNoAdvisor(self)
+        self.assertEquals(self.transcript.advisors.count(), 0)
+        user = User.objects.create_user(
+            username='rbiz',
+            password='k',
+            first_name='Rishi',
+            last_name='Biswas',)
+        advisor = Student.objects.create(user=user)
+        advisor.jhed = 'rbiswas4'
+        advisor.save()
+        data = {
+            'action': 'add',
+            'jhed': advisor.jhed,
+        }
+        request = self.factory.patch(
+            '/advising/forum/Fall/2019/', data=data, format='json')
+        response = get_response_for_semester(self, request, self.student.user)
+        self.assertEquals(response.status_code, status.HTTP_200_OK)
+        self.assertEquals(self.transcript.advisors.count(), 1)
+
+    def test_add_advisor_not_in_db(self):
+        setUpTranscriptNoAdvisor(self)
+        self.assertEquals(self.transcript.advisors.count(), 0)
+        data = {
+            'action': 'add',
+            'jhed': 'rbiswas4',
+        }
+        request = self.factory.patch(
+            '/advising/forum/Fall/2019/', data=data, format='json')
+        response = get_response_for_semester(self, request, self.student.user)
+        self.assertEquals(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEquals(self.transcript.advisors.count(), 0)
+
+    def test_add_advisor_already_exists(self):
+        setUpTranscript(self)
+        self.assertEquals(self.transcript.advisors.count(), 1)
+        data = {
+            'action': 'add',
+            'jhed': 'rbiswas4',
+        }
+        request = self.factory.patch(
+            '/advising/forum/Fall/2019/', data=data, format='json')
+        response = get_response_for_semester(self, request, self.student.user)
+        self.assertEquals(response.status_code, status.HTTP_200_OK)
+        self.assertEquals(self.transcript.advisors.count(), 1)
+
+    def test_remove_advisor(self):
+        setUpTranscript(self)
+        self.assertEquals(self.transcript.advisors.count(), 1)
+        data = {
+            'action': 'remove',
+            'jhed': self.advisor.jhed,
+        }
+        request = self.factory.patch(
+            '/advising/forum/Fall/2019/', data=data, format='json')
+        response = get_response_for_semester(self, request, self.student.user)
+        self.assertEquals(response.status_code, status.HTTP_200_OK)
+        self.assertEquals(self.transcript.advisors.count(), 0)
+
+    def test_remove_advisor_invalid(self):
```

Comment:
This might be better with a more specific test name - I can dig through the code to find out why the "remove" is invalid - but the naming should give me more of a hint so I don't have to. Maybe something like `test_remove_advisor_does_not_exist` if that's what it's testing.

Reply:
@rbiz4 This can probably be broken down into two separate tests, `test_remove_not_added_advisor` and `test_remove_nonexistent_advisor`